sdk/go/core/ydk/types/types.go

```diff
@@ -83,32 +85,170 @@ type AugmentCapabilitiesFunction func() map[string]string
 
 // Entity is a basic type that represents containers in YANG
 type Entity interface {
-	GetEntityPath(Entity) EntityPath
-	GetSegmentPath() string
+	GetSegmentPath() 				string
+	GetChildByName(string, string) 	Entity
 
-	HasDataOrFilter() bool
-
-	SetValue(string, string)
-	GetChildByName(string, string) Entity
-
-	GetChildren() map[string]Entity
+	GetChildren() 					map[string]Entity
+	GetLeafs()						map[string]interface{}
 
 	SetParent(Entity)
-	GetParent() Entity
+	GetParent() 					Entity
 
 	GetAugmentCapabilitiesFunction() AugmentCapabilitiesFunction
-	GetBundleYangModelsLocation() string
-	GetBundleName() string
+	GetBundleYangModelsLocation() 	string
+	GetBundleName() 				string
 
-	GetYangName() string
-	GetParentYangName() string
+	GetYangName() 					string
+	GetParentYangName() 			string
 
-	GetFilter() yfilter.YFilter
+	GetFilter() 					yfilter.YFilter
 }
 
 // Entity is a basic type that represents containers in YANG
 type Bits map[string]bool
 
+type BitsList struct {
+	Value []map[string]bool
+}
+
+func getGoName(name string) string {
```

Comment:
This function appears to be a duplicate of the [ydkgen code](https://github.com/CiscoDevNet/ydk-gen/blob/master/ydkgen/api_model.py#L227). So, if in the future, we make some change to the logic in this function, we should not forget to make the same change in the above ydkgen code. This could introduce potential bugs.

Could we avoid this duplication?

Instead of using this function to get the `goname`, should we be including the `goname` as another field in the `map` in the generated code itself?

Reply:
I don't think it's necessary since the naming convention won't change in the future and the goal was to reduce the generated code.

Reply:
naming does change :-) I'd recommend to remove any duplication

Reply:
It is not possible to remove this code and keep the same reduction in size in the generated code. Adding the go name in a map causes sanity.go to go from 2760 lines to 3186 and for the xr ipv4_bgp_oper.go to go from 1093458 to 1263119 lines.

Reply:
As discussed, this increase is ok with me. Thanks for making this change